lib/appscale_tools.py

```diff
@@ -636,14 +635,13 @@ def upload_app(cls, options):
         apps_result = json_result['apps']
         current_app = apps_result[app_id]
         http_port = current_app['http']
+        if not http_port:
+          continue
         break
       except ValueError:
         pass
       except KeyError:
-        pass
-      AppScaleLogger.verbose("Waiting {0} second(s) for a port to be assigned to {1}".\
-        format(sleep_time, app_id), options.verbose)
-      time.sleep(sleep_time)
+        time.sleep(20)
```

Comment:
It's not a path we expect to hit often and because it really depends on the AC cycle and getting info in the app_info_map, it makes sense to me to be 20.

Happy to reconsider if you think otherwise.